test/testdrive/kafka-avro-debezium-transaction.td

```diff
@@ -0,0 +1,365 @@
+
+
+$ kafka-ingest format=avro topic=data schema=${schema} timestamp=1
+{"before": null, "after": {"row": {"a": 8, "b": 9}}, "source": {"file": "binlog2", "pos": 1, "row": 1, "snapshot": {"boolean": false}}, "transaction": {"total_order": null, "id": "5"}}
+
+
+> CREATE SINK data_sink FROM data_schema_inline
+  INTO KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'testdrive-data-sink-${testdrive.seed}'
+  WITH (reuse_topic=true) FORMAT AVRO
+  USING CONFLUENT SCHEMA REGISTRY '${testdrive.schema-registry-url}'
+
+# Check that repeated Debezium messages are skipped.
+$ kafka-ingest format=avro topic=data schema=${schema} timestamp=1
+{"before": null, "after": {"row": {"a": 4, "b": 5}}, "source": {"file": "binlog", "pos": 1, "row": 1, "snapshot": {"boolean": false}}, "transaction": {"total_order": null, "id": "1"}}
+{"before": null, "after": {"row": {"a": 8, "b": 9}}, "source": {"file": "binlog2", "pos": 1, "row": 1, "snapshot": {"boolean": false}}, "transaction": {"total_order": null, "id": "5"}}
+
+> SELECT a, b FROM data_schema_inline
+a  b
+----
+1  1
+2  3
+4  5
+8  9
+
+# Now do data2
+$ kafka-ingest format=avro topic=data2 schema=${schema} timestamp=1
+{"before": null, "after": {"row": {"a": 101, "b": 101}}, "source": {"file": "binlog", "pos": 0, "row": 0, "snapshot": {"boolean": false}}, "transaction": {"total_order": null, "id": "1"}}
+
+> SELECT a, b FROM data2_schema_inline
+a    b
+---------
+101  101
+
+
+#
+# Create a source using a file schema
+#
+! CREATE MATERIALIZED SOURCE data_schema_file
+  FROM KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'testdrive-data-${testdrive.seed}'
+  FORMAT AVRO USING SCHEMA FILE 'data-schema.json'
+  ENVELOPE DEBEZIUM
+contains:No such file or directory
+
+$ file-append path=data-schema.json
+\${schema}
+
+> CREATE MATERIALIZED SOURCE data_schema_file
+  FROM KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'testdrive-data-${testdrive.seed}'
+  WITH(tx_metadata = data_txdata, tx_metadata_collection_name = 'testdrive-data-${testdrive.seed}')
+  FORMAT AVRO USING SCHEMA FILE '${testdrive.temp-dir}/data-schema.json'
+  ENVELOPE DEBEZIUM
+
+> SELECT a, b FROM data_schema_file
+a  b
+----
+1  1
+2  3
+4  5
+8  9
+
+$ set-regex match=\d{13} replacement=<TIMESTAMP>
+
+$ kafka-verify format=avro sink=materialize.public.data_sink sort-messages=true
+{"before": null, "after": {"row": {"a": 1, "b": 1}}, "transaction": {"id": "<TIMESTAMP>"}}
+{"before": null, "after": {"row": {"a": 2, "b": 3}}, "transaction": {"id": "<TIMESTAMP>"}}
+{"before": null, "after": {"row": {"a": 4, "b": 5}}, "transaction": {"id": "<TIMESTAMP>"}}
+
+$ kafka-verify format=avro sink=materialize.public.data_sink sort-messages=true
+{"before": null, "after": {"row": {"a": 8, "b": 9}}, "transaction": {"id": "<TIMESTAMP>"}}
+
+#
+# Test reading from the source when tx and data don't match
+#
+# Sleep to make sure we get a different RT timestamp from previous messages
+$ sleep-is-probably-flaky-i-have-justified-my-need-with-a-comment duration=5s
```

Comment:
I think we remove these sleeps and replace them with this which will wait exactly the right amount of time for the RT timestamp to progress:

```sql
> BEGIN
> DECLARE CURSOR c FOR TAIL data_schema_inline WITH (snapshot = false, progress = true)
> FETCH 1 c
mz_timestamp  mz_progressed  mz_diff  a       b
-------------------------------------------------
<timestamp>   true           <null>   <null>  <null>
> COMMIT
```

I believe testdrive runs with timestamp frequency set to 100ms